src/runtime_src/core/common/utils.cpp

```diff
@@ -259,4 +260,44 @@ parse_clock_id(const std::string& id)
   return clock_str != clock_map.end() ? clock_str->second : "N/A";
 }
 
+uint64_t
+mac_addr_to_value(std::string mac_addr)
+{
+  boost::erase_all(mac_addr, ":");
+  // Convert the reduced string into a value using the hex formatter
+  std::istringstream converter(mac_addr);
+  uint64_t mac_addr_value = 0;
+  converter >> std::hex >> mac_addr_value;
+  return mac_addr_value;
+}
+
+std::string
+value_to_mac_addr(const uint64_t mac_addr_value)
```

Comment:
It looks like this function is is designed to convert the given uint64_t value into a human readable mac address (e.g., 11:22:33:44:55).   If so, then it can be simplified as follows:

```
std::string
value_to_mac_addr(const uint64_t mac_addr_value)
{
  if (mac_addr_value & 0xFFFFFF0000000000) {
    std::string errMsg = boost::str(boost:format("Mac address exceed IP4 maximum value: 0x%x") % mac_addr_value);
    throw runtime_error(errMsg);
  }
  
  std::string formattedOutput = boost::str(boost::format("%02x:%02x:%02x:%02x:%02x") 
                                           % ((mac_addr_value >> (4 * 8)) & 0xFF)
                                           % ((mac_addr_value >> (3 * 8)) & 0xFF)
                                           % ((mac_addr_value >> (2 * 8)) & 0xFF)
                                           % ((mac_addr_value >> (1 * 8)) & 0xFF)
                                           % ((mac_addr_value >> (0 * 8)) & 0xFF));

  return formattedOutput;
}
```

Reply:
Neat! I did add an extra group to the mac address to keep it consistent with the previous versions (we display 6 groupings) and a matching change to the hex validation at the top of the function.